assests/javascript/index.js

```diff
@@ -1 +1,174 @@
-//
+const apiKey = "8cf248a4fd398e2908cf52a9f187375b";
+const mainSection = document.getElementById("main");
+
+getWeather();
+
+async function getWeather() {
+  try {
+    const city = await getCityLocation();
+    const curentWeather = await getCurrentWeather(city);
+    const dailyForecastingWeatherList = await getForecastingWeather(city);
+
+    showWeather(curentWeather, dailyForecastingWeatherList);
+  } catch {
+    showErrorMassage();
+  }
+}
+
+async function getCityLocation() {
+  const fetchedData = await fetch(
+    `https://api.openweathermap.org/geo/1.0/direct?q=Rasht,IR&appid=${apiKey}`
+  );
+  const cityList = await fetchedData.json();
+
+  return cityList[0];
+}
+
+async function getCurrentWeather(thisCity) {
+  const fetchedData = await fetch(
+    `https://api.openweathermap.org/data/2.5/weather?lat=${thisCity.lat}&lon=${thisCity.lon}&appid=${apiKey}`
+  );
+  return await fetchedData.json();
+}
+
+async function getForecastingWeather(thisCity) {
+  const fetchedData = await fetch(
+    `https://api.openweathermap.org/data/2.5/forecast?lat=${thisCity.lat}&lon=${thisCity.lon}&appid=${apiKey}`
+  );
+  const threeHoursForecastingWeatherFullInfo = await fetchedData.json();
+
+  return filterForecastingWeatherByDay(
+    threeHoursForecastingWeatherFullInfo.list
+  );
+}
+
+function showWeather(curentWeather, dailyForecastingWeatherList) {
+  mainSection.innerHTML += `<section class="city-section">
+    <h1 class="city-name">${curentWeather.name}</h1>
+    <time datetime="2020-02-14">${
+      convertTimeStampToDate(curentWeather.dt).dayOfWeek
+    }, ${convertTimeStampToDate(curentWeather.dt).dayOfMonth} ${
+    convertTimeStampToDate(curentWeather.dt).month
+  } ${convertTimeStampToDate(curentWeather.dt).year}</time>
```

Comment:
call this function once and store the value in a variable and then you can reference it. Multiple calls would end up multiple logic to run, which is unnecessary.

Reply:
like:
```js
const formattedDate = convertTimeStampToDate(curentWeather.dt);
```
And then: 
```js
formattedDate.dayOfWeek
formattedDate.month
// ...
```

Reply:
ok thankssss